scripts/ci/version_mgr.py

```diff
@@ -2,67 +2,126 @@
-            help="Get latest published version")
+                        help="Get latest published version")
+    parser.add_argument('-w', '--weekly', action="store_true",
+                        help="Mark as weekly")
     return parser.parse_args()
 
+
 def get_versions():
     data = None
-    url = 'https://testing.zephyrproject.org/daily_tests/versions.json'
-    urllib.request.urlretrieve(url, 'versions.json')
+    if not os.path.exists('versions.json'):
+        url = 'https://testing.zephyrproject.org/daily_tests/versions.json'
+        urllib.request.urlretrieve(url, 'versions.json')
     with open("versions.json", "r") as fp:
         data = json.load(fp)
 
     return data
 
+def handle_compat(item):
+    item_compat = {}
+    if isinstance(item, str):
+        item_compat['version'] =  item
+        item_compat['weekly'] = False
+        item_compat['date'] = None
+    else:
+        item_compat = item
+
+    return item_compat
+
 def show_versions():
     data = get_versions()
-    for v in data:
-        print(f"- {v}")
+    for item in data:
+        item_compat = handle_compat(item)
+        is_weekly = item_compat.get('weekly', False)
+        wstr = ""
+        datestr = ""
+        if is_weekly:
+            wstr = "(marked for weekly testing)"
+        if item_compat.get('date'):
+            datestr = f"published on {item_compat['date']}"
+        print(f"- {item_compat['version']} {datestr} {wstr}")
+
 
 def show_latest():
     data = get_versions()
-    print(data[-1])
+    latest = data[-1]
+    item_compat = handle_compat(latest)
+
+    ver = item_compat.get("version")
+    date = item_compat.get("date", False)
+    is_weekly = item_compat.get('weekly')
+    datestr = ""
+    if item_compat.get('date'):
+        datestr = f"published on {date}"
+    print(f"Latest version is {ver} {datestr}")
+    if is_weekly:
+        print("This version is marked for weekly testing.")
+
 
-def update(git_tree):
+def update(git_tree, is_weekly=False):
     g = Git(git_tree)
+    today = datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f')
     version = g.describe()
     published = False
     data = get_versions()
-    if version in data:
+
+    found = list(filter(lambda item: (isinstance(item, list) and
+                        item.get('version') == version) or item == version, data))
+    if found:
         published = True
         print("version already published")
     else:
         print(f"New version {version}, adding to file...")
 
     if data and not published:
         with open(VERSIONS_FILE, "w") as versions:
-            data.append(version)
+            item = {}
+            item['version'] = version
+            item['date'] = today
+            item['weekly'] = is_weekly
+            data.append(item)
             json.dump(data, versions)
 
 def main():
     args = parse_args()
     if args.update:
-        update(args.update)
+        update(args.update, args.weekly)
```

Comment:
Okay, that means weekly will be our default cadence.

Reply:
not sure what you mean with default. it is set on a weekly basis, the default is going to be still daily publication of the commit.